features/exploredash/src/main/java/org/dash/wallet/features/exploredash/ui/ExploreMapFragment.kt

```diff
@@ -192,82 +177,98 @@ class ExploreMapFragment: SupportMapFragment() {
+            map.moveCamera(CameraUpdateFactory.newLatLngBounds(latLngBounds, heightInPixel, heightInPixel, mapPadding))
+            lastFocusedUserLocation = mCurrentUserLocation
+        }
     }
 
-    private suspend fun loadMarkers(items: List<SearchResult>) {
+    private fun loadMarkers(items: List<SearchResult>) {
         val markerSize = resources.getDimensionPixelSize(R.dimen.explore_marker_size)
-        val exploreMarkerSize = resources.getDimensionPixelSize(R.dimen.merchant_marker_size)
 
-        futureTarget = items.map {
+        futureTarget.forEach { markersGlideRequestManager.clear(it) }
+        futureTarget = items.map { item ->
             markersGlideRequestManager
-                .asBitmap()
-                .load(it.logoLocation)
-                .placeholder(R.drawable.ic_merchant)
-                .apply(RequestOptions().centerCrop().circleCrop())
-                .listener(object : RequestListener<Bitmap> {
-                    override fun onLoadFailed(e: GlideException?, model: Any?, target: Target<Bitmap>?, isFirstResource: Boolean): Boolean {
-                        Log.i("GlideException" ,"${e?.message}")
-                        return false
-                    }
+                    .asBitmap()
+                    .load(item.logoLocation)
+                    .error(R.drawable.ic_merchant)
+                    .apply(RequestOptions().centerCrop().circleCrop())
+                    .listener(object : RequestListener<Bitmap> {
+                        override fun onLoadFailed(e: GlideException?, model: Any?, target: Target<Bitmap>?,
+                                                  isFirstResource: Boolean): Boolean {
+                            Log.i("GlideException","${e?.message}")
+                            if (item.latitude != null && item.longitude != null && markerDrawable != null) {
+                                lifecycleScope.launch {
+                                    addMarkerItemToMap(item.latitude!!, item.longitude!!, markerDrawable!!, item.id)
+                                }
+                            }
+                            return false
+                        }
 
-                    override fun onResourceReady(resource: Bitmap?, model: Any?, target: Target<Bitmap>?, dataSource: DataSource?, isFirstResource: Boolean): Boolean {
-                        Log.i(this@ExploreMapFragment::class.java.simpleName, "Resource loaded")
-                        return false
-                    }
-                })
-                .submit(markerSize, markerSize)
-        }.toMutableList()
-
-        val exploreMarkers = items.zip(futureTarget).map { pair ->
-            pair.first.latitude?.let {
-                pair.first.longitude?.let { it1 ->
-                    ExploreMarkerItemUI(
-                        pair.first.id,
-                        it, it1,
-                        try {
-                            pair.second.get()
```

Comment:
Calling `.get()` before rendering markers makes the map wait until all of the merchant/atm icons have been either resolved or failed to resolve. This can take several seconds with a slow connection. Better to add markers to the map incrementally as they are load or fail.